File: Programmers/Level2/가장큰수/가장큰수.py
def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key = lambda x: (x[0], x[1%len(x)], x[2%len(x)], x[3%len(x)]), reverse=True)
    return str(int(''.join(numbers)))

numbers = [3, 30, 34, 5, 9]

print(solution(numbers))

File: Programmers/Level2/가장큰수/가장큰수.js
function solution(numbers) {
  const maxNum = numbers
    .map((ele) => ele + "")
    .sort((a, b) => b + a - (a + b))
    .join("");
  return maxNum[0] === "0" ? "0" : maxNum;
}

const numbers = [6, 10, 2];
console.log(solution(numbers));
